R/ec2.R

```diff
@@ -0,0 +1,22 @@
+#' @import httr
+#' @importFrom jsonlist fromJson
+AWS_METADATA = "http://169.254.169.254/latest/meta-data/iam/security-credentials/" 
+
+#' @export
+ec2_roles <- function() {
+    r <- GET(AWS_METADATA)
+    out <- strsplit(content(r, "text"))
+    return(out)
+}
+
+#' @export
+get_ec2_role_credentials <- function(role) {
+    r <- GET(paste0(AWS_METADATA,role))
+    if (http_status(r)$category == "client_error") {
+        h <- headers(r)
+        out <- structure(content(r, "text"), headers = h, class = "aws_error")
```

Comment:
Error's from the endpoint are not JSON or XML, but XHTML, for an example with headers:

    HTTP/1.0 404 Not Found
    Content-Type: text/html
    Content-Length: 345
    Connection: close
    Date: Sun, 28 Aug 2016 18:12:22 GMT
    Server: EC2ws

    <?xml version="1.0" encoding="iso-8859-1"?>
    <!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
             "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
    <html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
     <head>
      <title>404 - Not Found</title>
     </head>
     <body>
      <h1>404 - Not Found</h1>
     </body>
    </html>